server/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```diff
@@ -627,21 +618,23 @@ public void messageReceived(UnicastPingRequest request, TransportChannel channel
 
     }
 
-    public static class UnicastPingRequest extends TransportRequest {
+    static class UnicastPingRequest extends TransportRequest {
 
-        int id;
-        TimeValue timeout;
-        PingResponse pingResponse;
+        final int id;
+        final TimeValue timeout;
+        final PingResponse pingResponse;
 
-        public UnicastPingRequest() {
+        UnicastPingRequest(int id, TimeValue timeout, PingResponse pingResponse) {
+            this.id = id;
+            this.timeout = timeout;
+            this.pingResponse = pingResponse;
         }
 
-        @Override
-        public void readFrom(StreamInput in) throws IOException {
```

Comment:
we should probably throw an UnsupportedOperationException here, as we should not use it.